ansible/roles/java/tasks/main.yml

```diff
@@ -1,8 +1,23 @@
 ---
+# Check the enable status of the Amazon Java package
+- name: Check Java package enabled status
+  shell: "amazon-linux-extras | grep -qE '([0-9]{,3})\\s{2}({{ amazon_extra_java_package }}=latest)' && echo -n INSTALLED || true"
+  register: amazon_extra
+  changed_when: false
+  when: ansible_distribution == "Amazon"
+
 # Enable the openjdk install via yum
 - name: Enable the JAVA amazon-linux-extras package
-  shell: "amazon-linux-extras enable java-openjdk11"
-  when: ansible_distribution == "Amazon"
+  command: "amazon-linux-extras enable {{ amazon_extra_java_package }}"
+  register: amazon_extra
+  when: ansible_distribution == "Amazon" and amazon_extra.stdout != 'INSTALLED'
+
+# Update the package cache
+- name: Update Package Cache (yum)
+  yum:
+    update_cache: yes
+  changed_when: false
+  when: ansible_pkg_mgr == "yum"
```

Comment:
This appears to be a duplicate task

Reply:
duplicate removed and variable names updated.